apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/valueset/ValueSetController.java

```diff
@@ -29,53 +31,65 @@
-        ValueSetStateChangeResponse response = valueSetStateManager.activateValueSet(codeSetNm);
-        return new ResponseEntity<>(response, null, response.getStatus());
-    }
-
-    @GetMapping
-    @PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
-    public Page<ValueSet> findAllValueSets(@PageableDefault(size = 25) Pageable pageable) {
-        return valueSetReador.findAllValueSets(pageable);
-    }
-
-    @PostMapping("search")
-    @PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
-    public Page<ValueSet> searchValueSearch(@RequestBody ValueSetSearchRequest search,
-            @PageableDefault(size = 25) Pageable pageable) {
-        return valueSetReador.searchValueSearch(search, pageable);
-    }
-
-    @GetMapping("{codeSetNm}/concepts")
-    @PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
-    public List<Concept> findConceptsByCodeSetName(@PathVariable String codeSetNm) {
-        return valueSetReador.findConceptCodes(codeSetNm);
-    }
+	private final ValueSetStateManager valueSetStateManager;
+
+	private final ValueSetReader valueSetReador;
+
+	private final ValueSetUpdater valueSetUpdater;
+
+	private final ValueSetCreator valueSetCreator;
+
+	private final UserDetailsProvider userDetailsProvider;
+
+	@PostMapping
+	@PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
+	public ResponseEntity<CreateValueSetResponse> createValueSet(@RequestBody ValueSetRequest request) {
+		Long userId = userDetailsProvider.getCurrentUserDetails().getId();
+		CreateValueSetResponse response = valueSetCreator.createValueSet(request, userId);
+		return new ResponseEntity<>(response, null, response.getStatus());
+
+	}
+
+	@DeleteMapping("/{codeSetNm}")
+	@PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
+	public ResponseEntity<ValueSetStateChangeResponse> deleteValueSet(@PathVariable String codeSetNm) {
+		ValueSetStateChangeResponse response = valueSetStateManager.deleteValueSet(codeSetNm);
+		return new ResponseEntity<>(response, null, response.getStatus());
+	}
+
+	@PatchMapping("/{codeSetNm}")
+	@PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
+	public ResponseEntity<ValueSetStateChangeResponse> activateValueSet(@PathVariable String codeSetNm) {
+		ValueSetStateChangeResponse response = valueSetStateManager.activateValueSet(codeSetNm);
+		return new ResponseEntity<>(response, null, response.getStatus());
+	}
+	
+	@GetMapping
+	@PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
+	public Page<ValueSet> findAllValueSets(@PageableDefault(size = 25) Pageable pageable)  {
+		return valueSetReador.findAllValueSets(pageable);
+		
+	}
+
+	@PostMapping("/update")
+	@PreAuthorize("hasAuthority('LDFADMINISTRATION-SYSTEM')")
+	public ResponseEntity<UpdatedValueSetResponse> updateValueSet(@RequestBody ValueSetUpdateRequest update) {
+		UpdatedValueSetResponse response = valueSetUpdater.updateValueSet(update);
+		return new ResponseEntity<>(response, null, response.getStatus());
+
+	}
```

Comment:
what if we updated the paths so that activate went to the `"{codeSetNm}/activate"` path and deactivate went to `"{codeSetNm}/deactivate"`? 

Then we could make the update request a PUT to `"{codeSetNm}"`


Thoughts?

Reply:
Sure I'll update. Update the Request to PUT and  append /activate  / deactivate to  the path.

Reply:
@mpeels  updated